uaa/src/test/java/org/cloudfoundry/identity/uaa/integration/feature/LoginIT.java

```diff
@@ -531,33 +531,65 @@ private void loginThroughDiscovery(String userEmail, String password) {
         IntegrationTestUtils.createZoneOrUpdateSubdomain(adminClient, baseUrl, zoneId, zoneId, testZone3.getConfig());
         webDriver.get(zoneUrl);
-        assertEquals(0, webDriver.findElements(By.xpath("//*[text()='Create account']")).size());
         assertEquals(0, webDriver.findElements(By.xpath("//*[text()='Reset password']")).size());
 
-        testZone3.getConfig().getLinks().setSelfService(new Links.SelfService().setSelfServiceLinksEnabled(true).setPasswd("/forgot_password").setSignup("http://example.com"));
+        testZone3.getConfig().getLinks().setSelfService(new Links.SelfService().setSelfServiceResetPasswordEnabled(true).setPasswd(null));
         IntegrationTestUtils.createZoneOrUpdateSubdomain(adminClient, baseUrl, zoneId, zoneId, testZone3.getConfig());
         webDriver.get(zoneUrl);
-        assertEquals(1, webDriver.findElements(By.xpath("//*[text()='Create account']")).size());
         assertEquals(1, webDriver.findElements(By.xpath("//*[text()='Reset password']")).size());
 
-        testZone3.getConfig().getLinks().setSelfService(new Links.SelfService().setSelfServiceLinksEnabled(true).setPasswd(null).setSignup(null));
+        testZone3.getConfig().getLinks().setSelfService(new Links.SelfService().setSelfServiceResetPasswordEnabled(true).setPasswd("/forgot_password"));
         IntegrationTestUtils.createZoneOrUpdateSubdomain(adminClient, baseUrl, zoneId, zoneId, testZone3.getConfig());
         webDriver.get(zoneUrl);
-        assertEquals(0, webDriver.findElements(By.xpath("//*[text()='Create account']")).size());
         assertEquals(1, webDriver.findElements(By.xpath("//*[text()='Reset password']")).size());
 
-        testZone3.getConfig().getLinks().setSelfService(new Links.SelfService().setSelfServiceLinksEnabled(true).setPasswd("/forgot_password").setSignup("/create_account"));
+        testZone3.getConfig().getLinks().setSelfService(new Links.SelfService().setSelfServiceResetPasswordEnabled(false).setPasswd("/forgot_password"));
+        IntegrationTestUtils.createZoneOrUpdateSubdomain(adminClient, baseUrl, zoneId, zoneId, testZone3.getConfig());
+        webDriver.get(zoneUrl);
+        assertEquals(0, webDriver.findElements(By.xpath("//*[text()='Reset password']")).size());
+    }
+
+    @Test
+    public void testSelfServiceCreateAccountLinksBehavior() {
+        RestTemplate adminClient = IntegrationTestUtils.getClientCredentialsTemplate(
+            IntegrationTestUtils.getClientCredentialsResource(baseUrl, new String[0], "admin", "adminsecret"));
+        String zoneId = "testzone3";
+        String zoneUrl = baseUrl.replace("localhost", zoneId+".localhost");
+        IdentityZone testZone3 = IntegrationTestUtils.createZoneOrUpdateSubdomain(adminClient, baseUrl, zoneId, zoneId, new IdentityZoneConfiguration());
+
+        testZone3.getConfig().getLinks().setSelfService(new Links.SelfService().setSelfServiceCreateAccountEnabled(true).setSignup(""));
```

Comment:
Minor. Each test case below calls `new Links.SelfService()...`. You could initialize it once at the beginning of the test and then re-use.

Reply:
Done